bit_manipulation/binary_exponentiation.cpp

```diff
@@ -0,0 +1,74 @@
+/**
+ * @file
+ * @brief Binary exponentiation for calculating power
+ * @details
```

Comment:
Provide a detailed description of the algorithm: what does it do, how does it work, etc..
Let us know if you need any help here or in our [**Discord community**](https://discord.gg/c7MnfGFGa6)! 😄